src/jit/morph.cpp

```diff
@@ -17132,9 +17132,12 @@ void Compiler::fgPromoteStructs()
                 // field accesses, but only block-level operations on the whole struct, if the struct
                 // has only one or two fields, then doing those block operations field-wise is probably faster
                 // than doing a whole-variable block operation (e.g., a hardware "copy loop" on x86).
-                // So if no fields are accessed independently, and there are three or more fields,
+                // Struct promotion also provides the following benefits: reduce stack frame size,
+                // reduce the need for zero init of stack frame, constant prop and copy prop.
+                // Asm diffs indicate that promoting structs up to 3 fields is a net size win.
```

Comment:
I have updated comments and also opened an issue (https://github.com/dotnet/coreclr/issues/9477) to track it